core/modules/catalog-next/store/category/mutations.ts

```diff
@@ -12,18 +13,14 @@ const mutations: MutationTree<CategoryState> = {
   },
   [types.CATEGORY_ADD_CATEGORY] (state, category: Category) {
     if (category) {
-      let newCategoryEntry = {}
-      newCategoryEntry[category.id] = category
-      state.categoriesMap = Object.assign(state.categoriesMap, newCategoryEntry)
+      Vue.set(state.categoriesMap, category.id, category)
     }
   },
   [types.CATEGORY_ADD_CATEGORIES] (state, categories: Category[] = []) {
     if (categories.length) {
-      let newCategoriesEntry = {}
       categories.forEach(category => {
-        newCategoriesEntry[category.id] = category
+        Vue.set(state.categoriesMap, category.id, category)
       })
-      state.categoriesMap = Object.assign(state.categoriesMap, newCategoriesEntry)
```

Comment:
i'll only change that back for one reason - with every category vue will mutate state and will invoke recalculate all computed properties and getters which are based on this. Fix for reactivity here is actually change
`state.categoriesMap = Object.assign(state.categoriesMap, newCategoriesEntry)`
to
`state.categoriesMap = Object.assign({}, state.categoriesMap, newCategoriesEntry)`

Reply:
Ah nice, didn't figured that out :)